OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/SensitiveWorkshopsServiceTests.cs

```diff
@@ -0,0 +1,224 @@
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins(filter).ConfigureAwait(false);
+
+        // Assert
+        resultExcpected.Should().BeEquivalentTo(result);
+    }
+
+    #endregion
+
+    #region With
+
+    private SearchResult<WorkshopDto> SetupFetchByFilterForAdmins(
+        string userId, bool isRegionAdmin, bool isMinistryAdmin, long parentCATOTTGId, RegionAdminDto adminRegion = null, MinistryAdminDto adminMinistry = null)
+    {
+        var workshops = WithListOfWorkshops().ToList();
+        var workshopsDto = WithListOfWorkshopsDto().ToList();
+        IEnumerable<long> subSettlementsIds = new List<long>() { parentCATOTTGId };
+        SetUpCurrentUserService(userId, isRegionAdmin, isMinistryAdmin);
+        SetupRegionAdminService(userId, adminRegion);
+        SetupMinistryAdminService(userId, adminMinistry);
+        SetUpCodeficatorService(subSettlementsIds, parentCATOTTGId);
+        SetUpWorkshopsRepository(workshops);
+        SetUpMapper(workshopsDto, workshops);
+        var resultExcpected = new SearchResult<WorkshopDto> { TotalAmount = workshopsDto.Count, Entities = workshopsDto };
+        return resultExcpected;
+    }
+
+    private IEnumerable<Workshop> WithListOfWorkshops()
+    {
+        var workshops = WorkshopGenerator.Generate(5);
+        return workshops;
+    }
+
+    private IEnumerable<WorkshopDto> WithListOfWorkshopsDto()
+    {
+        var workshops = WorkshopDtoGenerator.Generate(5);
+        return workshops;
+    }
+
+    private void SetUpCurrentUserService(string userId, bool isRegionAdmin = false, bool isMinistryAdmin = false)
+    {
+        currentUserServiceMock.Setup(u => u.IsRegionAdmin()).Returns(isRegionAdmin);
+        currentUserServiceMock.Setup(u => u.IsMinistryAdmin()).Returns(isMinistryAdmin);
+        currentUserServiceMock.Setup(u => u.UserId).Returns(userId);
+    }
+
+    private void SetupRegionAdminService(string userId, RegionAdminDto admin)
+    {
+        regionAdminServiceMock.Setup(a => a.GetByUserId(userId)).ReturnsAsync(admin);
+    }
+
+    private void SetupMinistryAdminService(string userId, MinistryAdminDto ministryAdmin)
+    {
+        ministryAdminServiceMock.Setup(a => a.GetByUserId(userId)).ReturnsAsync(ministryAdmin);
+    }
+
+    private void SetUpCodeficatorService(IEnumerable<long> subSettlementsIds, long parentId)
+    {
+        codeficatorServiceMock.Setup(c => c.GetAllChildrenIdsByParentIdAsync(parentId)).ReturnsAsync(subSettlementsIds);
+    }
+
+    private void SetUpWorkshopsRepository(List<Workshop> workshopsReturned)
+    {
+        workshopRepository.Setup(
+                w => w.Get(
+                    It.IsAny<int>(),
+                    It.IsAny<int>(),
+                    It.IsAny<string>(),
+                    It.IsAny<Expression<Func<Workshop, bool>>>(),
+                    It.IsAny<Dictionary<Expression<Func<Workshop, object>>, SortDirection>>(),
```

Comment:
(Testing wrong logic) Please avoid wherever possible usage of `It.IsAny()`, as it doesn't ensure that input parameters passed as expected. I.e. there're two first arguments of `int` type and they can be switched by mistake in real code, but your test will never catch this case.

Reply:
Thank you, i will fix this.